### PR TITLE
Add support for transforming keys and values to lowercase, uppercase o…

### PR DESCRIPTION
**Pull request by torse**
…r capitals.

This pull request adds two new configuration keys `transform_key` and `transform_value` that can be set to `lowercase`, `uppercase`, `capitalize` to transform either the keys or values. This should solve issue https://github.com/logstash-plugins/logstash-filter-kv/issues/24 and https://github.com/logstash-plugins/logstash-filter-kv/issues/16.

### CODE REVIEW

File: lib/logstash/filters/kv.rb [hunk 1]
```diff
@@ -60,6 +60,30 @@ class LogStash::Filters::KV < LogStash::Filters::Base
...
+  # Normalize values to lower case, upper case or capitals.
+  #
+  # For example, to capitalize all values:
+  # [source,ruby]
+  #     filter {
+  #       kv {
+  #         norm => "capitalize"
+  #       }
+  #     }
+  config :norm, :validate => ["lowercase", "uppercase", "capitalize"]
```
Comment (jordansissel) [inline lib/logstash/filters/kv.rb]: `norm` feels like the wrong name for this setting for two reasons. First, `norm` is abbreviated, and I prefer not to abbreviate setting names. Second, that I'm not sure `normalize' is the right name for the behavior. 

Maybe something like `key_letters` and `value_letters` for the setting names? Or `change_key` and `change_value` ?

```
key_letters => "lowercase"
# or this
change_key => "uppercase"
```

**Comment by jordansissel** [comment]
It looks like the CLA check failed. Can you read http://www.elasticsearch.org/contributor-agreement/ and sign it? We can proceed with review after that. :)

**Comment by torse** [comment]
I received the signed CLA via Email, just after I submitted the pull request. Maybe it just needs to be rechecked?

I've choosen `norm` because it is similar to `trim`, but I'am fine with changing it to any of your suggestions. Maybe also `convert_key`, `transform_key` or `modify_key` is an option? Please tell me what you prefer, I'll update the pull request and issue title accordingly.

Also, I was thinking of using constants instead strings in the ``case ... when`` check for performance reasons. Do you think it is worth it (I'm not an expert in Ruby)?

**Comment by jordansissel** [comment]
transform_key and transform_value I think are ok names


In terms of constants, it's a good idea, but I don't have any performance advice on that because it's not something I often test.

**Comment by jsvd** [comment]
tested, LGTM

**Pull request closed without merging**